arbor/include/arbor/domain_decomposition.hpp

```diff
@@ -47,17 +48,18 @@ class ARB_ARBOR_API domain_decomposition {
     int gid_domain(cell_gid_type gid) const;
     int num_domains() const;
     int domain_id() const;
+    cell_size_type index_on_domain(cell_gid_type gid) const;
```

Comment:
Possibly expose `std::pair<int, cell_size_type> gid_domain(cell_gid_type)`. This might allow skipping some duplicated calls to the inner function. Second, how about adding a proper type for the returned value?

Reply:
This would be a breaking API change. Internally, we never use them together, so there would be no direct benefit. 
But if you prefer, I can merge these functions.